doc/sphinx/proof-engine/ltac.rst

```diff
@@ -88,91 +88,72 @@ mode but it can also be used in toplevel definitions as shown below.
-                     : first [ `ltac_expr` | ... | `ltac_expr` ]
-                     : solve [ `ltac_expr` | ... | `ltac_expr` ]
-                     : idtac [ `message_token` ... `message_token`]
-                     : fail [`natural`] [`message_token` ... `message_token`]
-                     : gfail [`natural`] [`message_token` ... `message_token`]
-                     : fresh [ `component` … `component` ]
-                     : context `ident` [`term`]
-                     : eval `red_expr` in `term`
-                     : type of `term`
-                     : constr : `term`
-                     : uconstr : `term`
-                     : type_term `term`
-                     : numgoals
-                     : guard `test`
-                     : assert_fails `ltac_expr3`
-                     : assert_succeeds `ltac_expr3`
-                     : `tactic`
-                     : `qualid` `tacarg` ... `tacarg`
-                     : `atom`
-   atom              : `qualid`
-                     : ()
-                     : `int`
-                     : ( `ltac_expr` )
-   component : `string` | `qualid`
-   message_token     : `string` | `ident` | `int`
-   tacarg            : `qualid`
-                     : ()
-                     : ltac : `atom`
-                     : `term`
-   let_clause        : `ident` [`name` ... `name`] := `ltac_expr`
-   context_rule      : `context_hyp`, ..., `context_hyp` |- `cpattern` => `ltac_expr`
-                     : `cpattern` => `ltac_expr`
-                     : |- `cpattern` => `ltac_expr`
-                     : _ => `ltac_expr`
-   context_hyp       : `name` : `cpattern`
-                     : `name` := `cpattern` [: `cpattern`]
-   match_rule        : `cpattern` => `ltac_expr`
-                     : context [`ident`] [ `cpattern` ] => `ltac_expr`
-                     : _ => `ltac_expr`
-   test              : `int` = `int`
-                     : `int` (< | <= | > | >=) `int`
-   selector          : [`ident`]
-                     : `int`
-                     : (`int` | `int` - `int`), ..., (`int` | `int` - `int`)
-   toplevel_selector : `selector`
-                     : all
-                     : par
-                     : !
-
-.. productionlist:: coq
-   top              : [Local] Ltac `ltac_def` with ... with `ltac_def`
-   ltac_def         : `ident` [`ident` ... `ident`] := `ltac_expr`
-                    : `qualid` [`ident` ... `ident`] ::= `ltac_expr`
+.. insertprodn ltac_expr tactic_atom
+
+.. prodn::
+   ltac_expr ::= @binder_tactic
+   | @ltac_expr4
+   binder_tactic ::= fun {+ @name } => @ltac_expr
+   | let {? rec } @let_clause {* with @let_clause } in @ltac_expr
+   | info @ltac_expr
+   let_clause ::= @name := @ltac_expr
+   | @ident {+ @name } := @ltac_expr
+   ltac_expr4 ::= @ltac_expr3 ; @binder_tactic
+   | @ltac_expr3 ; @ltac_expr3
+   | @ltac_expr3 ; [ {? @multi_goal_tactics } ]
+   | @ltac_expr3 ; [> {? @multi_goal_tactics } ]
```

Comment:
For instance, these rules could be extracted as a `ltac_seq` non-terminal.

Reply:
Sure, of course they can be extracted.  But is that most usable presentation?